mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -451,6 +450,38 @@ pub async fn lp_init_continue(ctx: MmArc) -> MmInitResult<()> {
             .map_to_mm(MmInitError::ErrorSqliteInitializing)?;
         init_and_migrate_sql_db(&ctx).await?;
         migrate_db(&ctx)?;
+        // Initialize the global and wallet directories and databases since these are constants over the lifetime of KDF.
+        #[cfg(all(feature = "new-db-arch", not(target_arch = "wasm32")))]
+        {
+            use db_common::async_sql_conn::AsyncConnection;
+            use db_common::sqlite::rusqlite::Connection;
+            use std::sync::{Arc, Mutex};
+
+            let global_dir = ctx.db_root().join("global");
+            let wallet_dir = ctx.db_root().join("wallets").join(hex::encode(ctx.rmd160().as_slice()));
+            if !ensure_dir_is_writable(&global_dir) {
+                return MmError::err(MmInitError::db_directory_is_not_writable("global"));
+            };
```

Comment:
Thera are functions for these, you should use them here instead of code duplication.

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2378/commits/b4764889e4f1bf0bb7893755d8e752365fd09aa5